src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        stateCopy = {};
+        break;
+
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keyRemove of action.keysToRemove) {
+          if (stateCopy.hasOwnProperty(keyRemove)) {
+            delete stateCopy[keyRemove];
+          }
+        }
```

Comment:
```suggestion
          delete stateCopy[keyRemove];
        }
```
No need to add an additional check here, delete doesn't throw errors even if the property doesn't exist